shiny/render/_data_frame_utils/_datagridtable.py

```diff
@@ -0,0 +1,331 @@
+
+    Parameters
+    ----------
+    data
+        A pandas `DataFrame` object, or any object that has a `.to_pandas()` method
+        (e.g., a Polars data frame or Arrow table).
+    width
+        A _maximum_ amount of vertical space for the data table to occupy, in CSS units
+        (e.g. `"400px"`) or as a number, which will be interpreted as pixels. The
+        default is `fit-content`, which sets the table's width according to its
+        contents. Set this to `100%` to use the maximum available horizontal space.
+    height
+        A _maximum_ amount of vertical space for the data table to occupy, in CSS units
+        (e.g. `"400px"`) or as a number, which will be interpreted as pixels. If there
+        are more rows than can fit in this space, the table body will scroll. Set the
+        height to `None` to allow the table to grow to fit all of the rows (this is not
+        recommended for large data sets, as it may crash the browser).
+    summary
+        If `True` (the default), shows a message like "Viewing rows 1 through 10 of 20"
+        below the grid when not all of the rows are being shown. If `False`, the message
+        is not displayed. You can also specify a string template to customize the
+        message, containing `{start}`, `{end}`, and `{total}` tokens. For example:
+        `"Viendo filas {start} a {end} de {total}"`.
+    filters
+        If `True`, shows a row of filter inputs below the headers, one for each column.
+    editable
+        If `True`, allows the user to edit the cells in the grid. When a cell is edited,
+        the new value is sent to the server for processing. The server can then return
+        a new value for the cell, which will be displayed in the grid.
+    selection_mode
+        Single string to set the mode of the table.
+
+        Supported values:
+        * Use `"none"` to disable any cell selections or editing.
+        * Use `"row"` to allow a single row to be selected at a time.
+        * Use `"rows"` to allow multiple rows to be selected by clicking on them
+        individually.
+    row_selection_mode
+        Deprecated. Please use `mode={row_selection_mode}_row` instead.
+
+    Returns
+    -------
+    :
+        An object suitable for being returned from a `@render.data_frame`-decorated
+        output function.
+
+    See Also
+    --------
+    * :func:`~shiny.ui.output_data_frame`
+    * :class:`~shiny.render.data_frame`
+    * :class:`~shiny.render.DataGrid`
+    """
+
+    data: pd.DataFrame
+    width: str | float | None
+    height: str | float | None
+    summary: bool | str
+    filters: bool
+
+    def __init__(
+        self,
+        data: pd.DataFrame | PandasCompatible,
+        *,
+        width: str | float | None = "fit-content",
+        height: str | float | None = "500px",
+        summary: bool | str = True,
+        filters: bool = False,
+        editable: bool = False,
+        selection_mode: SelectionMode = "none",
```

Comment:
When we enable column and region selection mode, how will people specify that they want to enable some combination of these, for example, multi-row, multi-column, and region selection?

Reply:
`selection_mode` -> `selection_modes`